lib/temporal.rb

```diff
@@ -81,6 +83,41 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
+    # run_id of nil: await the latest run
+    def await_workflow_result(workflow:, workflow_id:, run_id: nil, **args)
```

Comment:
@antstorm I don't consider this highly specialized.  It'll be used in integration tests frequently, and sometimes in production.  In particular, this scenario will be more common than checking history.
A couple other points.
- The fact that this happens to use the underlying history GRPC endpoint is an implementation detail.  It could just as easily have been in "DescribeWorkflow" or a dedicated endpoint.  People would struggle to figure out that that's the right thing to call when wanting to check the result.
- The complexity of this method tells me that we wouldn't want developers to need to reimplement the logic.